src/pke/lib/scheme/bgvrns/bgvrns-parametergeneration.cpp

```diff
@@ -275,9 +276,11 @@ void ParameterGenerationBGVRNS::InitializeFloodingDgg(std::shared_ptr<CryptoPara
     usint r             = cryptoParamsBGVRNS->GetDigitSize();
     double B_e          = sqrt(alpha) * sigma;
     uint32_t auxBits    = DCRT_MODULUS::MAX_SIZE;
+    uint32_t thresholdParties = cryptoParamsBGVRNS->GetThresholdNumOfParties();
     // bound on the secret key is sigma*sqrt(alpha) if the secret is sampled from discrete gaussian distribution
-    // and is 1 if the secret is sampled from ternary distribution.
-    double Bkey = (cryptoParamsBGVRNS->GetSecretKeyDist() == GAUSSIAN) ? sigma * sqrt(alpha) : 1;
+    // and is 1 * threshold number of parties if the secret is sampled from ternary distribution. The threshold number of
+    // parties is 1 by default but can be set to the number of parties in a threshold application.
+    double Bkey = (cryptoParamsBGVRNS->GetSecretKeyDist() == GAUSSIAN) ? sigma * sqrt(alpha) : 1 * thresholdParties;
```

Comment:
you don't need this multiplication by 1, an assignment would suffice. "1 * thresholdParties;" can be tranformed to a comment